pandas/core/common.py

```diff
@@ -225,14 +225,27 @@ def count_not_none(*args) -> int:
     return sum(x is not None for x in args)
 
 
-def asarray_tuplesafe(values, dtype: NpDtype | None = None) -> np.ndarray:
+@overload
+def asarray_tuplesafe(
+    values: ArrayLike | list | tuple | zip, dtype: NpDtype | None = ...
```

Comment:
Alternatively to special-casing, we could omit `zip` here and have the caller convert the argument to a list first, instead of having this function do the conversion (currently there is one call site using `zip`).

Reply:
I'm fine with having `zip` here.